awake.lua

```diff
@@ -45,35 +45,78 @@ local g = grid.connect()
 
 local alt = false
 
-mode = 1
-mode_names = {"STEP","LOOP","SOUND","OPTION"}
+local mode = 1
+local mode_names = {"STEP","LOOP","SOUND","OPTION"}
 
-one = {
+local one = {
   pos = 0,
   length = 8,
   start = 1,
   data = {1,0,3,5,6,7,8,7,0,0,0,0,0,0,0,0}
 }
-two = {
+
+local two = {
   pos = 0,
   length = 7,
   start = 1,
   data = {5,7,0,0,0,0,0,0,0,0,0,0,0,0,0,0}
 }
 
+function add_pattern_params() 
+  params:add_separator()
+  
+  params:add{type = "number", id = "one_length", name = "<one> length]", min=1, max=8, 
+    default = one.length,
+    action=function(x) one.length = x end }
+
+  params:add{type = "number", id = "one_start", name = "<one> start]", min=1, max=16, 
+    default=one.start,
+    action=function(x) one.start = x end }
+  
+  for i=1,16 do
+    params:add{type = "number", id= ("one_data_"..i), name = ("<one> data "..i), min=1, max=8, 
+      default = one.data[i],
+      action=function(x)one.data[i] = x end }
+  end
+  
+  params:add_separator()
+  
+  params:add{type = "number", id = "two_length", name = "<two> length]",  min=1, max=8, 
+    default = two.length,
+    action=function(x)two.length = x end}
+  
+  params:add{type = "number", id = "two_start", name = "<two> start]",  min=1, max=16, 
+    default = two.start,
+    action=function(x)two.start = x end }
+  
+  for i=1,16 do
+    params:add{type = "number", id= "two_data_"..i, name = "<two> data "..i,  min=1, max=8, 
+      default = two.data[i],
+      action=function(x) two.data[i] = x end }
+  end
+  
+  params:add_separator()
+end
+
+local set_loop_data = function(which, step, val)
+  params:set(which.."_data_"..step, val)
+end
+
+
 local midi_out_device
 local midi_out_channel
 
 local scale_names = {}
-notes = {}
+local notes = {}
 local active_notes = {}
 
 local edit_ch = 1
 local edit_pos = 1
 
 snd_sel = 1
```

Comment:
Should this also be local?

Reply:
actually, with recent changes to reset global enviroment between scripts (https://github.com/monome/norns/pull/929), better practice would be to make _none_ of these top-level variables local! (they will be nuked when the next script is launched, and keeping them global makes them accessible from maiden REPL.) so, my bad.

i think all the "common" patches need a good cleanup to bring in line with current recommended best practices.

(e.g. in this script, the declaration of control specs is probably redundant.)